src/main/java/com/cinema/dao/impl/OrderDaoImpl.java

```diff
@@ -0,0 +1,51 @@
+package com.cinema.dao.impl;
+
+import com.cinema.dao.OrderDao;
+import com.cinema.exceptions.DataProcessingException;
+import com.cinema.lib.Dao;
+import com.cinema.model.Order;
+import com.cinema.model.User;
+import com.cinema.util.HibernateUtil;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class OrderDaoImpl implements OrderDao {
+
+    public Order create(Order order) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.persist(order);
+            transaction.commit();
+            return order;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert to DB Order: " + order.toString(), e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<Order> getOrderHistory(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<Order> query = session.createQuery("from Order o join fetch o.tickets "
+                    + "where o.user.id = :id", Order.class);
```

Comment:
You can also add JOIN FETCH for User here so that the Hibernate doesn't create an extra query for it. And you can fetch user directly, without looking for the user id.
```suggestion
            Query<Order> query = session.createQuery("FROM Order o JOIN FETCH o.tickets "
                    + "JOIN FETCH o.user "
                    + "where o.user = :user", Order.class);
```

Reply:
I agree :)

Reply:
done